### PR TITLE
Postgres and Redshift: support `ALTER SCHEMA`

### LINKED ISSUE #3320: redshift dialect: missing support for ALTER TABLE
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

Redshift supports `ALTER TABLE` statement according to their [documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_SCHEMA.html).

sqlfluff fails while parsing files containing this statement.

### Expected Behaviour

Add support for the `ALTER SCHEMA` statements.

### Observed Behaviour

Not currently supported, results in parse errors.
```
== [scripts/redshift/ddl/schema_name/core_schema.sql] FAIL                                                                                                                                                                                                                      
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'ALTER SCHEMA schema_name OWNER TO redshift_...'
```

### How to reproduce

Create a test file containing as below and try to lint/parse with sqlfluff.
```
CREATE SCHEMA IF NOT EXISTS schema_name;
ALTER SCHEMA schema_name OWNER TO redshift_user;
```

### Dialect

redshift

### Version

`sqlfluff, version 0.13.1`
`Python 3.8.10`

### Configuration

```
# https://docs.sqlfluff.com/en/stable/configuration.html#config

[sqlfluff]
dialect = redshift
templater = jinja
rules = all
exclude_rules = none
# Controls SQLFluff output, see max_line_length for SQL output
output_line_length = 118
ignore_templated_areas = true
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = true

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 118
indent_unit = space
comma_style = trailing
# Whether or not to allow a single element in the select clause to be without an alias


# https://docs.sqlfluff.com/en/stable/rules.html#ruleref

[sqlfluff:rules:L003]
hanging_indents = true

[sqlfluff:rules:L007]
operator_new_lines = after

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper

[sqlfluff:rules:L011]
# of tables
aliasing = explicit

[sqlfluff:rules:L012]
# of columns
aliasing = explicit

[sqlfluff:rules:L013]
allow_scalar = false

[sqlfluff:rules:L029]
unquoted_identifiers_policy = all

[sqlfluff:rules:L031]
# Avoid table aliases in from clauses and join conditions
force_enable = true

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]
capitalisation_policy = upper

[sqlfluff:rules:L051]
fully_qualify_join_types = both

[sqlfluff:rules:L052]
multiline_newline = false
require_final_semicolon = false

[sqlfluff:rules:L054]
group_by_and_order_by_style = explicit

[sqlfluff:rules:L057]
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = false

[sqlfluff:rules:L059]
prefer_quoted_identifiers = false

[sqlfluff:rules:L063]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by mdahlman**
### Brief summary of the change made

fixes #3320

### Are there any other side effects of this change that we should be aware of?
I don't think so.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - I created `postgres_alter_schema.sql`
  - I generated `postgres_alter_schema.yml` and validated it

### Full disclosure
I was fully successful with unit tests like these:
```
pytest -s test/dialects/dialects_test.py -k postgres_alter_schema.sql]
tox -e py38 -- -s test/dialects/dialects_test.py -k postgres_alter_schema.sql]
```

But I had some errors in this test:
```
tox -e py38
...
Results (364.35s):
    3743 passed
      40 failed
         - test/diff_quality_plugin_test.py:13 test_diff_quality_plugin[linter/diffquality/parse_error.sql-expected_violations_lines2]
         - test/cli/commands_test.py:63 test__cli__command_directed
         - test/cli/commands_test.py:139 test__cli__command_lint_stdin[command0]
         - test/cli/commands_test.py:139 test__cli__command_lint_stdin[command1]
...
```
It seemed to me that this is an issue with my environment rather than my code... but I'm not certain how to confirm this.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 1]
```diff
@@ -4170,3 +4171,30 @@ class AlterTypeStatementSegment(BaseSegment):
             ),
         ),
     )
+
+
+class AlterSchemaStatementSegment(BaseSegment):
+    """An `ALTER SCHEMA` statement.
+
+    https://www.postgresql.org/docs/current/sql-alterschema.html
+
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_postgres.py:4180]: Nit

```suggestion
```

File: test/fixtures/dialects/postgres/postgres_alter_schema.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER SCHEMA schema1 RENAME TO schema2;
+ALTER SCHEMA schema1 OWNER TO new_owner;
```
Comment (tunetheweb) [inline test/fixtures/dialects/postgres/postgres_alter_schema.sql:2]: Nit:

```suggestion
ALTER SCHEMA schema1 OWNER TO new_owner;

```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
As the original ask was for RedShift could we add the same test case to `test/fixtures/dialects/redshift`? Just to make sure it doesn't revert in future for some reason.

File: test/fixtures/dialects/postgres/postgres_alter_schema.sql (hunk 2 above)
Comment (mdahlman) [reply test/fixtures/dialects/postgres/postgres_alter_schema.sql:2]: To use git properly am I supposed to **_accept_** this suggestion somehow? or should I **_make_** this change?
Obviously it's extremely easy for me to add the new line... I just want to be sure that I'm using git correctly.

Comment (mdahlman) [reply test/fixtures/dialects/postgres/postgres_alter_schema.sql]: Nevermind. I see how to commit your suggestion now.

Comment (tunetheweb) [reply test/fixtures/dialects/postgres/postgres_alter_schema.sql]: You can do either!

For simple changes like this you can use the `commit suggestion` button above.
To accept multiple suggestions, you can batch the acceptances and then commit in one go (quicker than accepting individually if lots of them).
Either of those will resolve this conversation automatically.
Note you'll need to pull these changes from GitHub to your local branch if you want to make any more changes locally.

However, if there's lots of other changes than can do offline and push. At which point these conversations will show as "outdated". you can then flip through all outdated ones and resolve them if happy they outdating did indeed resolve the conversation.

**Comment by mdahlman** [comment]
Sounds good. I couldn't think of a reason to make the Redshift SQL different from the PostgreSQL SQL. So they are currently identical with my latest commit.

**Comment by tunetheweb** [comment]
> Sounds good. I couldn't think of a reason to make the Redshift SQL different from the PostgreSQL SQL. So they are currently identical with my latest commit.

SGTM!

**Comment by tunetheweb** [comment]
I've a few other PRs to merge so happy to keep this open if you're going to look at it in next day or two. If not done by Friday's planned release then will merge this.

**Review by tunetheweb — APPROVED** [review]
LGTM!

**Pull request merged**